File: src/main/java/com/engineering/challenges/maths/CountDigits.java
package com.engineering.challenges.maths;

public class CountDigits {
    public static void main(String[] args) {
        int n = 12345;

        countDigits1(n);

        countDigits2(n);

        countDigits3(n);
    }

    private static void countDigits3(int n) {
        int length = (int) Math.floor(Math.log10(n) + 1);

        System.out.println(length);
    }

    private static void countDigits2(int n) {
        String s = String.valueOf(n);

        System.out.println(s.length());
    }

    private static void countDigits1(int n) {
        int count = 0;
        while (n != 0) {
            n = n / 10;
            count++;
        }

        System.out.println(count);
    }
}

File: js-iv/src/countdigits.ts
// 1
function countDigits(n: number): number {
    if (typeof n === 'number') {
        return String(n).length             // TC: O(n)
    }
    else return -1
}
console.log(countDigits(12433))

// 2
function countDigits1(n: number): number {
    let count = 0
    while (n > 0) {
        count++
        n = Math.floor(n / 10)              // TC: O(log10 N)
    }
    return count
}
console.log(countDigits1(12345))

// 3
function countDigits2(n: number): number {
    return Math.floor(Math.log10(n) + 1)    // TC: O(1)
}
console.log(countDigits2(12345))